Sentence transformers in the Hub

osanseviero opened the pull request:
There are still a couple of TODOs and some hacks. Here are some screenshots

<img width="777" alt="Screen Shot 2021-06-15 at 4 08 13 PM" src="https://user-images.githubusercontent.com/7246357/122068132-6c712200-cdf4-11eb-9fa3-6ade9bd6f4cf.png">
<img width="726" alt="Screen Shot 2021-06-15 at 4 08 20 PM" src="https://user-images.githubusercontent.com/7246357/122068136-6e3ae580-cdf4-11eb-8d63-f527d452788b.png">
<img width="652" alt="Screen Shot 2021-06-15 at 4 08 27 PM" src="https://user-images.githubusercontent.com/7246357/122068145-6f6c1280-cdf4-11eb-826b-e55ddac4b31c.png">
<img width="702" alt="Screen Shot 2021-06-15 at 4 08 31 PM" src="https://user-images.githubusercontent.com/7246357/122068154-71ce6c80-cdf4-11eb-8063-a7bb2656650e.png">
<img width="746" alt="Screen Shot 2021-06-15 at 4 08 37 PM" src="https://user-images.githubusercontent.com/7246357/122068158-72ff9980-cdf4-11eb-89c4-e5a7ce66a86d.png">
<img width="696" alt="Screen Shot 2021-06-15 at 4 08 42 PM" src="https://user-images.githubusercontent.com/7246357/122068166-74c95d00-cdf4-11eb-9de2-11071daea25d.png">
<img width="714" alt="Screen Shot 2021-06-15 at 4 08 45 PM" src="https://user-images.githubusercontent.com/7246357/122068169-75fa8a00-cdf4-11eb-83c2-c545de5669a8.png">

osanseviero commented:
@patrickvonplaten in case you have a free cycle to take a look :)

patrickvonplaten commented on sentence-transformers-in-the-hub.md:
```diff
@@ -0,0 +1,144 @@
+    </a>
+</div>
+
+Over the past few weeks, we've built collaborations with many Open Source frameworks in the machine mearning ecosystem. One that gets us particularly excited is Sentence Transformers.
```
```suggestion
Over the past few weeks, we've built collaborations with many Open Source frameworks in the machine learning ecosystem. One that gets us particularly excited is Sentence Transformers.
```

patrickvonplaten commented on sentence-transformers-in-the-hub.md:
```diff
@@ -0,0 +1,144 @@
+
+Over the past few weeks, we've built collaborations with many Open Source frameworks in the machine mearning ecosystem. One that gets us particularly excited is Sentence Transformers.
+
+[Sentence Transformers](https://github.com/UKPLab/sentence-transformers) is a framework for sentence, paragraph and image embeddings. This allows to derive semantically meaningful embeddings (1) which is useful for very interesting applications, such as semantic search or mutli-lingual zero shot classification. As part of Sentence Transformers [v2 release](TODO add link), there are a lot of cool  ...
```
I'd remove the "very interesting" part.

osanseviero replied:
Done :)

patrickvonplaten commented on sentence-transformers-in-the-hub.md:
```diff
@@ -0,0 +1,144 @@
+- Sharing your models in the 🤗 Hub easily.
+- Widgets and Inference API for sentence embeddings and sentence similarity.
+- Better sentence-embeddings models available ([benchmark](https://www.sbert.net/docs/pretrained_models.html#sentence-embedding-models) and [models](https://huggingface.co/sentence-transformers) in the 🤗 Hub).
+- Learning to rank (LTR) functions for cross-encoder training.
```
Maybe a link here that explains what LTR is?

nreimers replied:
I decided to move the LTR functions to version 2.1

I want to make a big special on LTR with educational material, which LTR methods exists, how they can be used and which LTR method yields the best result. But this requires still some time & experiments, hence, I moved it to version 2.1 of sentence-transformers.

osanseviero replied:
Thanks, I just removed that point

osanseviero commented:
@gary149 do you think you could help us with a thumbnail for this and other sentence transformers announcements? It could be something like Sentence Transformers + Hugging Face like we've done for other collaborations.

CC @julien-c in case you want to check out the blog post. Sorry for the ugly hacks for the widget, it's some technical debt that I think it's ok until we improve a bit the widgets situation. 😅 

@nreimers I updated the sentence similarity widget to use a sentence similarity widget. I also changed to the new `save_to_hub` method. I think everything is pretty much ready, we're just missing the link to the actual release 😄

nreimers commented:
@osanseviero 
Here the link to the release:
https://github.com/UKPLab/sentence-transformers/releases/tag/v2.0.0

Release notes will be added tomorrow.

julien-c commented on sentence-transformers-in-the-hub.md:
```diff
@@ -0,0 +1,141 @@
+headers = {"Authorization": "Bearer YOUR_TOKEN"}
+
+def query(payload):
+	data = json.dumps(payload)
+	response = requests.request("POST", API_URL, headers=headers, data=data)
+	return json.loads(response.content.decode("utf-8"))
```
```suggestion
	response = requests.post(API_URL, headers=headers, json=payload)
	return response.json()
```

julien-c replied:
(yeah i know we need to update the inference API modal, not sure why it was written like this for the first place)

julien-c commented on sentence-transformers-in-the-hub.md:
```diff
@@ -0,0 +1,141 @@
+
+## What's next?
+
+Moving forward, we want to make this integration even more useful. In our roadmap, we expect training and evaluation data to be included in the automatically created model card.
```
```suggestion
Moving forward, we want to make this integration even more useful. In our roadmap, we expect training and evaluation data to be included in the automatically created model card, like is the case in `transformers` from version `v4.8`.
```

julien-c commented:
looks good!

> Sorry for the ugly hacks for the widget, it's some technical debt that I think it's ok until we improve a bit the widgets situation. 😅

I think that's totally fine. We have a way to render (just) the widget server side so we could have hacked together something outside of the .md, but I think this works

LysandreJik commented:
:tada:
